### DISCUSSION TITLE
SqliteError: unable to open database file

### DISCUSSION THREAD

**Discussion by Chugarah** [epicweb-dev/epic-stack#879] [Q&A]
Hi, my firs time posting about an issue, I like what I have seen so far with features, documentation and the dedicated creator Kent C. Dodds :).

I am using Vscode with node js in windows. Created an new folder, opened in Vscode and enterted this command in the terminal

npx create-epic-app@latestclear

The build process looks beatiful with alot of stuff happening but getting an error when trying to do something with the database, after some small investigation, I think the problem could be with one of the test folder that is using "import Database from 'better-sqlite3'" instead of prism and errors out

-------------Output---------------------
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.17.0) to .\node_modules\@prisma\client in 69ms

Start by importing your Prisma Client (See: http://pris.ly/d/importing-client)

Tip: Easily identify and fix slow SQL queries in your app. Optimize helps you enhance your visibility: https://pris.ly/--optimize

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQLite database "data.db" at "file:./prisma/data.db?connection_limit=1"

SQLite database data.db created at file:./prisma/data.db?connection_limit=1

1 migration found in prisma/migrations

Applying migration `20230914194400_init`

The following migration(s) have been applied:

migrations/
  └─ 20230914194400_init/
    └─ migration.sql

All migrations have been successfully applied.
Environment variables loaded from .env
Running seed command `tsx prisma/seed.ts` ...
🌱 Seeding...
SqliteError: unable to open database file
    at new Database (C:\Users\MeMe\OneDrive - Core\HTML och CSS\silicon-ab-2\node_modules\better-sqlite3\lib\database.js:69:26)
    at cleanupDb (C:\Users\MeMe\OneDrive - Core\HTML och CSS\silicon-ab-2\tests\db-utils.ts:142:13)
    at seed (C:\Users\Meme\OneDrive - Core\HTML och CSS\silicon-ab-2\prisma\seed.ts:20:8)
    at <anonymous> (C:\Users\Meme\OneDrive - Core\HTML och CSS\silicon-ab-2\prisma\seed.ts:222:1) {
  -------------Output---------------------
  
  I am still an beginner and not sure why this problem should accour on fresh install.
  Thanks for feedback/Input how to solve this.

**Comment by kentcdodds** [comment]
Hello! I'm not sure what could be causing this issue for you, but I was actually just looking at that and I was thinking we could probably do away with that function completely now. If someone wants to have a fresh seed of their database they can run a prisma database reset.

Try removing that function call and the function itself and see if that gets things working for you. Then I think I'll probably remove it as well.

**Comment by Jabril-Mahamud** [comment]
Hi there,

I'm still experiencing this issue. I heard from someone in your Discord community that switching to macOS resolved the problem for them. Unfortunately, I only have access to Windows.

Any kind of support would appreciated. Thank you! :)

PS I tried removing it, no change :(

**Comment by Jabril-Mahamud** [comment]
Hello again, I've got good news! The npx create-epic-app@latest works now! The only changes I made was that I reset my computer and I was given the following lines that I didn't see the first time when I think I should. Given this message, it seems like there was a seeding issue with Prisma. I've never used it so I can't really give much help on this. I even deployed successfully.

Context: Fresh windows PC. The cli had to install chromium and stuff as well.
Below is a picture, I hope I could help.

![Screenshot 2024-11-05 145006](https://github.com/user-attachments/assets/85fe6162-a5f8-41bb-ac97-20e008961ed0)

**Reply by Jabril-Mahamud** [reply]
The only odd thing I found it seems that there's no users? Not even Kody :(

![image](https://github.com/user-attachments/assets/1c89a057-f38a-4010-9d29-5a5e992c3fc7)

**Reply by kentcdodds** [reply]
I wonder if the prisma reset command is somehow creating a different database from the one that's used when running the app. Could you look for a `data.db` file in your repo directory and see whether there are more than one?

**Comment by Jabril-Mahamud** [comment]
I only have 1 data.db. I've cleared the cache and am trying again. I'll let
you know if it works.

**Comment by Jabril-Mahamud** [comment]
Locally, I have data in the db. But in the deployed version I have nothing.

**Reply by kentcdodds** [reply]
> But in the deployed version I have nothing.

I'm not sure I understand what you mean by this.

**Comment by Jabril-Mahamud** [comment]
Sorry, if I understand properly. The deployed version on fly.io should also
have certain data already in like users, and notes by default? I do not.
I thank you for spending your precious time helping me. I followed the
instructions to a T and I have also tried to run multiple deployments. I've
also followed your instructions of deleting cache. Here is the repo I am
having issues with.

https://github.com/Jabril-Mahamud/LifeOS


[image: image.png]

**Reply by kentcdodds** [reply]
> The deployed version on fly.io should also have certain data already in like users, and notes by default?

No, it should not have anything by default other than what's in the migrations. You wouldn't want to have fake data in production.

**Comment by Jabril-Mahamud** [comment]
Sorry
;-;
